### ISSUE TITLE
Complex pattern matching

### ISSUE THREAD

**Issue by zaoqi** [vlang/v#1249]
Will V support complex pattern match like racket?
https://docs.racket-lang.org/reference/match.html

**Comment by benman1** [comment]
Isn't the [match](https://docs.racket-lang.org/reference/match.html) keyword clearer?

**Comment by ntrel** [comment]
It's hard to follow the Racket syntax as it's based on Lisp. Here's Rust:
https://doc.rust-lang.org/1.9.0/book/patterns.html

I'm not sure we need things like destructuring (yet), or perhaps only in limited cases like for tuples. Giving a matched expression a new short name might be useful sometimes.

**Comment by lutherwenxu** [comment]
What features do V currently need (compared to the current `match`)?

**Comment by penguindark** [comment]
`match` has improved a lot from this discussion.
Old issue no more actual.
